inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+
+#include "transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.hpp"
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SimplifyCTCGreedyDecoderSeqLen, "SimplifyCTCGreedyDecoder", 0);
+
+ngraph::pass::SimplifyCTCGreedyDecoderSeqLen::SimplifyCTCGreedyDecoderSeqLen() {
+    MATCHER_SCOPE(SimplifyCTCGreedyDecoderSeqLen);
+    auto decoder = pattern::wrap_type<opset6::CTCGreedyDecoderSeqLen>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto decoder_seq_len = std::dynamic_pointer_cast<opset6::CTCGreedyDecoderSeqLen> (m.get_match_root());
+        if (!decoder_seq_len) {
+            return false;
+        }
+
+        if (decoder_seq_len->get_input_size() > 2) {
+            const auto seq_len_pshape = decoder_seq_len->get_input_partial_shape(0);
+            auto blank_index = std::dynamic_pointer_cast<ngraph::opset6::Constant>(decoder_seq_len->input_value(2).get_node_shared_ptr());
+            if (!blank_index || seq_len_pshape.rank().is_dynamic() || seq_len_pshape[2].is_dynamic()) {
```

Comment:
You need also check that `seq_len_pshape.rank().get_length() is > 2`

Reply:
No need. According to documentation input data of CTCGreedyDecoderSeqLen have layout [N,T,C] so seq_len_pshape.rank().get_length() always will be == 3